### PR TITLE
Update 06/2020

### PR DESCRIPTION
**Pull request by 3imed-jaberi**
- update CI pipeline (10+)
- update .gitignore file
- fix README.md file
- lint with xo over native eslint
- improve the src code
- improve the test code
- add license file
- improve pkg.json

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,130 +15,137 @@ const colorCodes = {
...
-        transporter(str, args)
-      } else {
-        console.log(...args)
-      }
-    }
-  }())
-
-  return async function logger (ctx, next) {
+    const transporter =
+      typeof options === 'function'
```
Comment (dead-horse) [inline index.js:24]: I'd better to keep if else which is more readable.

File: index.js [hunk 2]
```diff
@@ -28,130 +15,137 @@ const colorCodes = {
...
-
-/**
- * Log helper.
- */
+  };
+};
 
-function log (print, ctx, start, len, err, event) {
+// Log helper.
+// eslint-disable-next-line max-params
```
Comment (dead-horse) [inline index.js:93]: I don't like the lint rule, it's too restrictive, and change the code style is useless to this module.

Comment (3imed-jaberi) [reply index.js:93]: I will add some rules to XO config ...

**Review by dead-horse — APPROVED** [review]

**Comment by dead-horse** [comment]
ops. I think we missed the README file.

**Pull request merged**